examples/Components/Dropdown/Dropdown.purs

```diff
@@ -2,100 +2,104 @@ module Docs.Components.Dropdown where
 
 import Prelude
 
-import Effect.Aff.Class (class MonadAff)
 import Data.Array (difference, mapWithIndex)
+import Data.Const (Const)
 import Data.Maybe (Maybe(..))
+import Data.Variant (SProxy(..), onMatch)
+import Docs.CSS as CSS
+import Effect.Aff.Class (class MonadAff)
 import Halogen as H
 import Halogen.HTML as HH
-import Halogen.HTML.Events as HE
 import Halogen.HTML.Properties as HP
 import Select as Select
+import Select.Simple as SS
 import Select.Setters as Setters
-import Docs.CSS as CSS
 
 type State =
   { items :: Array String
-  , text :: String }
+  , text :: String 
+  }
 
 type Input =
-  { items :: Array String }
+  { items :: Array String 
+  }
 
-data Query a
-  = HandleSelect (Select.Message Query String) a
+data Action
+  = HandleSelect (SS.Message String)
 
-data Message
-  = Void
+type Query = Const Void
+type Message = Void
 
-type ChildSlot = Unit
-type ChildQuery = Select.Query Query String
+type ChildSlots m = 
+  ( select :: SS.Slot String m Unit
+  )
+
+_select = SProxy :: SProxy "select"
 
 component :: ∀ m. MonadAff m => H.Component HH.HTML Query Input Message m
-component =
-  H.parentComponent
-    { initialState
-    , render
-    , eval
-    , receiver: const Nothing
-    }
+component = H.mkComponent
+  { initialState
+  , render
+  , eval: H.mkEval $ H.defaultEval { handleAction = handleAction }
+  }
   where
-    initialState :: Input -> State
-    initialState i = { items: i.items, text: "Select an option" }
+  initialState :: Input -> State
+  initialState i = { items: i.items, text: "Select an option" }
 
-    eval :: Query ~> H.ParentDSL State Query ChildQuery ChildSlot Message m
-    eval = case _ of
-      HandleSelect (Select.Selected item) a -> do
-        st <- H.get
-        _ <- H.query unit $ Select.setVisibility Select.Off
-        _ <- H.query unit $ Select.replaceItems $ difference st.items [ item ]
-        H.modify_ _ { text = item }
-        pure a
+  handleAction :: Action -> H.HalogenM State Action (ChildSlots m) Message m Unit
+  handleAction = case _ of
+    HandleSelect msg -> onMatch
+      { selected: \item -> do
+          st <- H.get
+          _ <- H.query _select unit $ H.tell $ Select.PerformAction $ Select.setVisibility Select.Off
+          _ <- H.query _select unit $ H.tell $ Select.ReplaceItems $ difference st.items [ item ]
```

Comment:
Hmm, if we update our helpers then less of this has to change
```purescript
setVisibility = H.tell <<< PerformAction <<< SetVisibility
```
```purescript
_ <- H.query _select unit $ Select.setVisibility Select.Off
```

Reply:
But that would preclude the use of `setVisibility` in the render function, which is of type `H.ComponentHTML Action ChildSlots m`. You can't use queries in components anymore.

An alternative is to make everything a query, and then have an action `PerformQuery (Query Unit)` that lets you use the action-style ones. But then we get back into the extensibility problem.